semmc-learning/src/SemMC/Stochastic/Initialize.hs

```diff
@@ -82,10 +85,10 @@ data Config arch =
          }
 
 -- | Synthesis environment.
-data SynEnv t solver fs arch =
-  SynEnv { seFormulas :: STM.TVar (MapF.MapF ((A.Opcode arch) (A.Operand arch)) (F.ParameterizedFormula (Sym t solver fs) arch))
+data SynEnv solver t fs arch =
```

Comment:
What is the motivation for swapping the order of the `solver` and `t` type parameters here? It's the cause of a fair bit of downstream code churn, but I'm not sure why it's necessary.

Reply:
I'm trying to keep things in sync with the parameter order of the `OnlineBackend`.  I think it's really confusing when things like this don't match.

Reply:
I'm aligned. I've been trying to fix that kind of thing myself